File: 12 - Recursão/base2.c
#include <stdio.h>

int base2 (int i) {
    if (i < 2)
        return i;
    return base2(i / 2) * 10 + (i % 2);
}

int main() {
    int i;

    printf("This program converts a number from base 10 to base 2.\nType a non-negative integer value: and press ENTER: ");
    if (!scanf(" %d", &i) || i < 0) {
        printf("Invalid value passed.\n");
        return 1;
    }
    printf("The decimal %d is equivalent to the binary %d in base 2.\n", i, base2(i));
}

File: 12 - Recursão/base2.py
import sys


def base2(i):
    if i < 2:
        return i % 2
    return base2(i // 2) * 10 + (i % 2)


def main():
    print("This program converts a number from base 10 to base 2.")

    try:
        n = int(input("Type a non-negative integer value: "))
    except ValueError:
        print("Error: not a number.")
        sys.exit()
    except EOFError:
        print("")
        sys.exit()
    if n < 0:
        print("Error:", n, "is negative.")
        sys.exit()
    print("The decimal", n, "is equivalent to the binary", base2(n), "in base 2.")


main()
